javagen/src/main/java/com/azure/autorest/mapper/ModelPropertyMapper.java

```diff
@@ -111,16 +124,11 @@ public ClientModelProperty map(Property property) {
         builder.headerCollectionPrefix(headerCollectionPrefix);
 
         IType propertyWireType = Mappers.getSchemaMapper().map(property.getSchema());
-        if (propertyWireType != null && property.isNullable()) {
-            propertyWireType = propertyWireType.asNullable();
-        }
-
-        IType propertyClientType = Mappers.getSchemaMapper().map((property.getSchema())).getClientType();
-
-        if (!property.isRequired()) {
-            propertyClientType = propertyClientType.asNullable();
+        if (property.isNullable() || !property.isRequired()) {
             propertyWireType = propertyWireType.asNullable();
         }
+        // Invariant: clientType == wireType.getClientType()
+        IType propertyClientType = propertyWireType.getClientType();
```

Comment:
This part of logic is slightly different. In condition of `property.isNullable()`, previously it is only `propertyWireType` be nullable, now both is nullable (I assume when property is nullable, the client type of that property should also be nullable?).